cassdegrees/ui/forms.py

```diff
@@ -0,0 +1,74 @@
+from django import forms
+from django.forms import ModelForm
+from api.models import DegreeModel, SubplanModel
+
+
+class EditProgramFormSnippet(ModelForm):
+    class Meta:
+        model = DegreeModel
+        fields = ('code', 'year', 'name', 'units', 'degreeType')
+        widgets = {
+            'code': forms.TextInput(attrs={'class': "text tfull"}),
+            'year': forms.TextInput(attrs={'class': "text tfull", 'type': "number"}),
+            'name': forms.TextInput(attrs={'class': "text tfull"}),
+            'units': forms.TextInput(attrs={'class': "text tfull", 'type': "number"}),
```

Comment:
This I *do* like, however. Can we adapt this to the entire project? Seems bad to have this duplicated in different places everywhere.

Reply:
Does this do validation of any kind/restrictions on editing code/etc?

Reply:
It can be adapted for anything with a model, including instantiating blank forms for creation. We can either add validation through django forms built in is_valid(), or overlaying javascript like you've done in other areas. Using this approach should hugely cut down duplication, and the associated html generation.

Reply:
Fair. This is a required feature of the underlying user stories for creation (and by extension, editing), so I would suggest trying Pristine or whatever just to clean this up - should be fairly trivial.

Reply:
At the moment it rejects all the required use cases with the exception of program(name, year) as this one doesn't have a uniqueness constraint in the DB. Again, something we should talk about tomorrow. Happy to implement further validation, but we should be taking a common approach across the board.